### PR TITLE
EFGS ACC ready

### PR DESCRIPTION
**Pull request by jendakol**

### CODE REVIEW

File: internal/functions/efgs/api/structs.go [hunk 1]
```diff
@@ -117,11 +117,16 @@ type DiagnosisKeyWrapper struct {
 
 //ToData convert struct from DiagnosisKeyWrapper to DiagnosisKey
 func (wrappedKey *DiagnosisKeyWrapper) ToData() *DiagnosisKey {
+	rt := wrappedKey.TransmissionRiskLevel
+	if rt > 8 || rt < 0 {
+		rt = 0 // This is an override for putting 0x7fffffff there
+	}
+
```
Comment (marekaf) [inline internal/functions/efgs/api/structs.go]: why is this needed? can we document it a bit better in the code?

Comment (jendakol) [reply internal/functions/efgs/api/structs.go]: This is a sad story of how EFGS has put this into their protocol and then recommended to not use it because the value meaning can be different across different countries. Because it _cannot_ be unused in fact, the final recommendation is to put max value possible there. That works for others but, because we put those keys into the Key server, not for us - the Key server doesn't like it so we have to adjust the value to basically anything _consumable_. The value itself is not used at all for anything and even though it's said to be _optional_ in comments, the Key server requires it and fails when the value is not there or is invalid.  
Sad story.

Comment (jendakol) [reply internal/functions/efgs/api/structs.go]: I can surely put this novel above into the code if you like @marekaf

Comment (marekaf) [reply internal/functions/efgs/api/structs.go]: yes please 👍

File: internal/functions/efgs/upload-batch.go [hunk 2]
```diff
@@ -104,47 +104,32 @@ func UploadBatch(w http.ResponseWriter, r *http.Request) {
...
-	var err error
-	var config uploadConfiguration
+	efgsEnv := efgsutils.GetEfgsEnvironmentOrFail()
 
-	config.NBTLSPair, err = efgsutils.LoadX509KeyPair(ctx, efgsutils.NBTLS)
-	if err != nil {
-		logger.Debug("Error loading authentication certificate")
-		return nil, err
-	}
+	url := efgsutils.GetEfgsURLOrFail(efgsEnv)
```
Comment (marekaf) [inline internal/functions/efgs/upload-batch.go]: can we please use Dependency Injection everywhere? We use it in like half of the cases and the codebase is very inconsistent. This approach of mutating the internal state inside of the functions based on global vars or calling other functions is impossible to unit test. 

I'd recommend reading like the beginning of this blogpost for example https://medium.com/@elliotchance/a-new-simpler-way-to-do-dependency-injection-in-go-9e191bef50d5

Comment (jendakol) [reply internal/functions/efgs/upload-batch.go]: This is impossible to unit test anyway. There are so many things missing for that goal that I don't believe this one makes any difference...  
There's a function named `uploadBatch` below and it receives the configuration (that is created here on **this** line)
```go
type uploadConfiguration struct {
	URL       *urlutils.URL
	Env       efgsutils.Environment
	NBTLSPair *efgsutils.X509KeyPair
	Client    *http.Client
	BatchTag  string
}

func uploadBatch(ctx context.Context, batch *efgsapi.DiagnosisKeyBatch, config *uploadConfiguration) (*efgsapi.UploadBatchResponse, error) { ... }
```
Even `Client` is there. The only thing that is untestable here is database handling - ok, I can put even the `Database` instance inside of the `uploadConfiguration`.

Comment (jendakol) [reply internal/functions/efgs/upload-batch.go]: I know how to create testable code. It's just very hard to do such thing in a language like Go. The linked article didn't convince me about anything but the Go is very screwed language for writing complex applications.

Comment (marekaf) [reply internal/functions/efgs/upload-batch.go]: Our application is far from complex. We use Go and there is nothing we can do about it anyway. What we can do however is use DI and write testable functions.

Comment (jendakol) [reply internal/functions/efgs/upload-batch.go]: Yes, it's very far, yet complex enough :smile:

Comment (jendakol) [reply internal/functions/efgs/upload-batch.go]: The whole EFGS code was refactored to be more testable.

**Review by marekaf — APPROVED** [review]
lgtm!! thanks!
